File: leetcode/0001 Two Sum/two_sum.go
func twoSum(nums []int, target int) []int {
    noMap := make(map[int]int)
    for index, no := range nums {
        if i, ok:= noMap[target-no]; ok {
            return []int{i, index}
        }
        noMap[no] = index
    }
    return nil
}

File: leetcode/0001 Two Sum/TwoSum.java
class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int index = 0; index < nums.length; index++) {
            if (map.containsKey(nums[index]))
                return new int[]{map.get(nums[index]), index};
            else
                map.put(target - nums[index], index);
        }
        return new int[]{-1, -1};
    }
}